### ISSUE TITLE
OutputKeys.CDATA_SECTION_ELEMENTS not work in jdk 1.8.0.332.b09

### ISSUE THREAD

**Issue by swingfly** [adoptium/adoptium-support#529]
## Summary

OutputKeys.CDATA_SECTION_ELEMENTS doesn't work after updating jdk from 1.8.0.322.b06 to 1.8.0.332.b09.

## Steps to reproduce
`TransformerFactory transformerFactory = TransformerFactory.newInstance();`
 `Transformer transformer = transformerFactory.newTransformer();`
 ` transformer.setOutputProperty(OutputKeys.CDATA_SECTION_ELEMENTS, "description");`

## Expected results
`<description><![CDATA[<es:thumbnail></es:thumbnail><es:details>123<br><p>tttt</p></es:details>]]></description>`

## Actual results
`<description>&lt;es:thumbnail&gt;&lt;/es:thumbnail&gt;&lt;es:details&gt;123&lt;/es:details&gt;</description>`

## Triaging info

Java version: jdk 1.8.0.332.b09

What is your operating system and platform?

MacOS

How did you install Java?
Download pkg from this github.

Did it work before?

It works well in JDK 1.8.0.322.b06

**Comment by karianna** [comment]
https://bugs.openjdk.org/browse/JDK-8288972

**Comment by jerboaa** [comment]
@swingfly I'm not able to reproduce this with:

```

import java.io.ByteArrayInputStream;
import java.io.InputStream;
import java.io.StringReader;
import java.io.StringWriter;
import java.nio.charset.StandardCharsets;
import javax.xml.transform.Transformer;
import javax.xml.transform.TransformerException;
import javax.xml.transform.TransformerFactory;
import javax.xml.transform.stream.StreamResult;
import javax.xml.transform.stream.StreamSource;

import org.junit.Assert;
import org.junit.Test;
import javax.xml.transform.OutputKeys;

/*
 * @test
 * @run junit/othervm CdataSectionTransform
 * @summary Verifies that a OutputKeys.CDATA_SECTION_ELEMENTS is adhered to correctly
 * @bug 8288972
 */
public class CdataSectionTransform {
    final String xsl =
        "<xsl:stylesheet version=\"1.0\" xmlns:xsl=\"http://www.w3.org/1999/XSL/Transform\">\n" +
        "    <xsl:output omit-xml-declaration=\"yes\" indent=\"no\" />\n" +
        "\n" +
        "    <xsl:template match=\"/\">\n" +
	"    <description>\n" +
        "        <xsl:value-of select=\"/source\" />\n" +
	"    </description>\n" +
        "    </xsl:template>\n" +
        "</xsl:stylesheet>\n";

    @Test
    public final void testTransformWithCdataKey() throws Exception {
        String[] xmls = prepareXML();
        Transformer t = createTransformerFromInputstream(
                new ByteArrayInputStream(xsl.getBytes(StandardCharsets.UTF_8)));
        t.setOutputProperty(OutputKeys.ENCODING, StandardCharsets.UTF_8.name());
	t.setOutputProperty(OutputKeys.CDATA_SECTION_ELEMENTS, "description");
        StringWriter sw = new StringWriter();
        t.transform(new StreamSource(new StringReader(xmls[0])), new StreamResult(sw));
	String actual = sw.toString().replaceAll(System.lineSeparator(), "\n");
        Assert.assertEquals(xmls[1], actual);
    }

    private String[] prepareXML() {
        String xml = "<?xml version=\"1.0\" encoding=\"UTF-8\"?><source>&lt;es:thumbnail&gt;&lt;/es:thumbnail&gt;&lt;es:details&gt;123&lt;/es:details&gt;</source>";
        String expected = "<description><![CDATA[<es:thumbnail></es:thumbnail><es:details>123</es:details>]]></description>";
        return new String[]{xml, expected};
    }

    private Transformer createTransformerFromInputstream(InputStream xslStream)
            throws TransformerException {
        return TransformerFactory.newInstance().newTransformer(new StreamSource(xslStream));
    }
}
```

**Comment by swingfly** [comment]
Our implementation looks likes this, working with JAXB,

```
import org.junit.Test;

import javax.xml.bind.JAXBContext;
import javax.xml.bind.JAXBException;
import javax.xml.bind.Marshaller;
import javax.xml.bind.annotation.XmlAccessType;
import javax.xml.bind.annotation.XmlAccessorType;
import javax.xml.bind.annotation.XmlElement;
import javax.xml.bind.annotation.XmlRootElement;
import javax.xml.parsers.DocumentBuilderFactory;
import javax.xml.parsers.ParserConfigurationException;
import javax.xml.transform.OutputKeys;
import javax.xml.transform.Transformer;
import javax.xml.transform.TransformerException;
import javax.xml.transform.TransformerFactory;
import javax.xml.transform.dom.DOMSource;
import javax.xml.transform.stream.StreamResult;
import java.io.ByteArrayOutputStream;
import java.io.IOException;
import java.util.List;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotNull;

@XmlRootElement(name = "kml")
@XmlAccessorType(XmlAccessType.FIELD)
public class Kml {
    private static final JAXBContext context;

    static {
        try {
            context = JAXBContext.newInstance(Kml.class);
        } catch (JAXBException e) {
            throw new RuntimeException(e);
        }
    }

    @XmlElement(name = "Document")
    protected Document document;

    public Kml() {
        document = new Document();
    }

    public Document getDocument() {
        return document;
    }

    public void setDocument(Document document) {
        this.document = document;
    }

    public List<Placemark> getPlacemarks() {
        return document.getFolder().getPlacemarks();
    }

    public List<Style> getStyles() {
        return document.getStyles();
    }

    public Placemark createPlaceMark() {
        Placemark placemark = new Placemark();
        document.getFolder().addPlaceMarker(placemark);
        return placemark;
    }

    public Style createStyle() {
        Style style = new Style();
        document.addStyle(style);
        return style;
    }

    public Style createIgnorableStyle() {
        Style style = new Style();
        document.getIgnorable().add(style);
        return style;
    }

    public String toXmlString() throws TransformerException, JAXBException, ParserConfigurationException, IOException {
        try (ByteArrayOutputStream outputStream = new ByteArrayOutputStream()) {
            Marshaller marshaller = context.createMarshaller();

            // Marshal xml into a empty dom
            DocumentBuilderFactory docBuilderFactory = DocumentBuilderFactory.newInstance();
            org.w3c.dom.Document document = docBuilderFactory.newDocumentBuilder().newDocument();

            marshaller.marshal(this, document);

            // transform the output, set "description" tag as a CDATA field
            TransformerFactory transformerFactory = TransformerFactory.newInstance();
            Transformer transformer = transformerFactory.newTransformer();
            transformer.setOutputProperty(OutputKeys.CDATA_SECTION_ELEMENTS, "description");

            transformer.transform(new DOMSource(document), new StreamResult(outputStream));
            return outputStream.toString();
        } catch (IOException | JAXBException | ParserConfigurationException | TransformerException e) {
            throw e;
        }
    }


    @Test
    public void testMarshal() throws IOException, JAXBException, TransformerException, ParserConfigurationException {
        Kml kml = new Kml();
        Style style = kml.createStyle();
        style.setId("123");
        style.getSubStyles().add(getIconStyle());

        Style style2 = kml.createStyle();
        style2.setId("124");
        style2.getSubStyles().add(getLineStyle());

        Placemark route = kml.createPlaceMark();
        route.setStyleUrl("#123");
        route.setName("test placemark");
        route.setId("789");
        route.setLineString(getLineString());
        route.setWayPoints(getLineString());

        Placemark placemark2 = kml.createPlaceMark();
        placemark2.setId("123");
        placemark2.setName("test placemark");
        placemark2.setDescription("<es:thumbnail></es:thumbnail><es:details>123<br><p>tttt</p></es:details>");
        placemark2.setStyleUrl("#123");
        placemark2.setPoint(getPoint());

        Placemark placemark3 = kml.createPlaceMark();
        placemark3.setId("345");
        placemark3.setName("test distancemark");
        placemark3.setDescription("<es:thumbnail></es:thumbnail><es:details>123</es:details>");
        placemark3.setStyleUrl("#124");
        placemark3.setPoint(getPoint());
        placemark3.setDistanceMarker(true);

        String str = kml.toXmlString();
        assertNotNull(str);
        String expect = "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"no\"?><kml xmlns=\"http://www.opengis.net/kml/2.2\"><Document><Style id=\"123\"><IconStyle><scale>0.96</scale><Icon><href>http://photos-images.active.com/file/3/1/original/9a10211f-ebc9-4585-b8eb-c1614e718406.png</href></Icon><hotSpot x=\"0.5\" xunits=\"fraction\" y=\"-0.25\" yunits=\"fraction\"/></IconStyle></Style><Style id=\"124\"><LineStyle><color>ff0000FF</color><width>5</width></LineStyle></Style><Ignorable/><Folder><Placemark id=\"789\"><name>test placemark</name><styleUrl>#123</styleUrl><LineString><tessellate>1</tessellate><coordinates>-0.14522,50.837645,0.0 -0.143375,50.83669600000001,0.0</coordinates></LineString><WayPoints><tessellate>1</tessellate><coordinates>-0.14522,50.837645,0.0 -0.143375,50.83669600000001,0.0</coordinates></WayPoints></Placemark><Placemark id=\"123\"><name>test placemark</name><description><![CDATA[<es:thumbnail></es:thumbnail><es:details>123<br><p>tttt</p></es:details>]]></description><styleUrl>#123</styleUrl><Point><altitudeMode>clampToGround</altitudeMode><coordinates>-0.14522,50.837645</coordinates></Point></Placemark><Placemark distanceMarker=\"true\" id=\"345\"><name>test distancemark</name><description><![CDATA[<es:thumbnail></es:thumbnail><es:details>123</es:details>]]></description><styleUrl>#124</styleUrl><Point><altitudeMode>clampToGround</altitudeMode><coordinates>-0.14522,50.837645</coordinates></Point></Placemark></Folder></Document></kml>";
        assertEquals(expect, str);
    }

    private LineString getLineString() {
        LineString lineString = new LineString();
        lineString.setCoordinates("-0.14522,50.837645,0.0 -0.143375,50.83669600000001,0.0");

        return lineString;
    }

    private LineStyle getLineStyle() {
        return new LineStyle("ff0000FF", 5);
    }
    
    private Point getPoint() {
        return new Point("-0.14522,50.837645");
    }

    private IconStyle getIconStyle() {
        IconStyle iconStyle = new IconStyle();
        iconStyle.setHotSpot(getHotSpot(0.5, -0.25, "fraction", "fraction"));
        iconStyle.setScale(0.96);
        iconStyle.setIcon(new Icon("http://link.googd,job/file/3/1/original/9a10211f-ebc9-4585-b8eb-c1614e718406.png"));

        return iconStyle;
    }

    private HotSpot getHotSpot(double x, double y, String xunits, String yunits) {
        return new HotSpot(x, y, xunits, yunits);
    }
}
```

**Comment by swingfly** [comment]
And I found there are some change making isCdataSection to return false in this PR **8041523: Xerces Update: Serializer improvements from Xalan**

**Comment by jerboaa** [comment]
I'm sorry, but I'm not able to reproduce it. Even if I use a JAXB generated XML. Please try to create a minimal reproducer (with no external jaxb implementations and external deps if possible) and mention the precise steps one needs to do to reproduce this. The example above doesn't comilpile for me. What I do notice is that you are using a namespace in your target XML and you might need to set the namespace via `setOutputProperty`:
https://docs.oracle.com/javase/8/docs/api/javax/xml/transform/Transformer.html#setOutputProperty-java.lang.String-java.lang.String-

**Comment by swingfly** [comment]
Will fix this issue?
Or is there any suggestion to bypass this ?

**Comment by swingfly** [comment]
Ok, I will downgrade my java. Thx.

**Comment by swingfly** [comment]
Updated.
I try to set CDATA property with namespace to solve the issue above temporarily as https://docs.oracle.com/javase/8/docs/api/javax/xml/transform/Transformer.html#setOutputProperty-java.lang.String-java.lang.String- :
```
transformer.setOutputProperty(OutputKeys.CDATA_SECTION_ELEMENTS, "{http://www.opengis.net/kml/2.2}description");
```
But it doesn't work yet.
I found the namespace was always parsed to "//www.opengis.net/kml/2.2" in method setCdataSectionElements of ToStream through debug.
I'm not sure this is another bug of jdk.

This is our namespace setting in package-info.java

```
@XmlSchema(
        namespace = "http://www.opengis.net/kml/2.2",
        elementFormDefault = javax.xml.bind.annotation.XmlNsForm.QUALIFIED
)
import javax.xml.bind.annotation.XmlSchema;
```
